Sources/XCTest/XCTestCase.swift

```diff
@@ -92,7 +92,7 @@ extension XCTestCase {
                 let testCase = self.init()
                 testCase.name = "\(testCase.dynamicType).\(name)"
 
-                var failures = [XCTFailure]()
+                var failures: [XCTFailure] = []
```

Comment:
I've seen [style guides that argue the former syntax is preferable](https://github.com/raywenderlich/swift-style-guide#type-inference). The convention in swift-corelibs-xctest also appears to be the former syntax. I don't have a strong preference either way, but I do think it's better to keep whatever convention is already being used, rather than make a large change such as this one.